nwb_conversion_tools/utils/json_schema.py

```diff
@@ -31,33 +31,50 @@ def append_replace_dict_in_list(d, ls, k):
     """
     if k in d and len(ls) > 0:
         # Index where the value dictionary[k] exists in the list of dicts
-        ind = np.where([d[k] == i[k] for i in ls])[0]
-        if len(ind) > 0:
-            ls[ind[0]] = d
+        indxs = np.where([d[k] == i[k] for i in ls])[0]
+        if len(indxs) > 0:
+            for idx in indxs:
+                ls[idx] = dict_deep_update(ls[idx], d)
         else:
             ls.append(d)
     else:
         ls.append(d)
     return ls
 
 
-def dict_deep_update(d: dict, u: dict, append_list: bool = True, remove_repeats: bool = True) -> dict:
+def dict_get_dtype(d: collections.abc.Mapping, key: str, default_val=None):
+    ret_val = d.get(key, default_val)
+    if isinstance(ret_val, type(default_val)) or default_val is None:
+        return ret_val
+    else:
+        return default_val
+
+
+def dict_deep_update(
+    d: collections.abc.Mapping,
+    u: collections.abc.Mapping,
+    append_list: bool = True,
+    remove_repeats: bool = True,
+    copy: bool = False,
```

Comment:
I think its good to have this as an optional arg, since dicts are mutable, its updating may not be desired.